cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,869 @@
+ *     to the {@code Ready} state when the rebalancing is done.
+ *
+ *     The user is also able to stop an in-progress rebalance proposal computation or an actual rebalancing,
+ *     annotating the custom resource with {@code strimzi.io/rebalance=stop} when it is in {@code PendingProposal}
+ *     or {@code Rebalancing} state; the operator moves to the {@code Stopped} state and the user can request a new
+ *     proposal applying the {@code strimzi.io/rebalance=refresh} annotation on the custom resource.
+ *
+ *     Finally, when a proposal is ready but it is stale because the user haven't approve it right after the
+ *     computation, so that the cluster conditions could be change, he can refresh the proposal annotating
+ *     the custom resource with the {@code strimzi.io/rebalance=refresh} annotation.
+ * </p>
+ */
+public class KafkaRebalanceAssemblyOperator
+        extends AbstractOperator<KafkaRebalance, AbstractWatchableResourceOperator<KubernetesClient, KafkaRebalance, KafkaRebalanceList, DoneableKafkaRebalance, Resource<KafkaRebalance, DoneableKafkaRebalance>>> {
+
+    private static final Logger log = LogManager.getLogger(KafkaRebalanceAssemblyOperator.class.getName());
+
+    // this annotation with related possible values (approve, stop, refresh) is set by the user for interacting
+    // with the rebalance operator in order to start, stop or refresh rebalacing proposals and operations
+    public static final String ANNO_STRIMZI_IO_REBALANCE = Annotations.STRIMZI_DOMAIN + "rebalance";
+    private static final long REBALANCE_POLLING_TIMER_MS = 5_000;
+    private static final int MAX_API_RETRIES = 5;
+
+    private final CrdOperator<KubernetesClient, KafkaRebalance, KafkaRebalanceList, DoneableKafkaRebalance> kafkaRebalanceOperator;
+    private final CrdOperator<KubernetesClient, Kafka, KafkaList, DoneableKafka> kafkaOperator;
+    private final PlatformFeaturesAvailability pfa;
+    private final Function<Vertx, CruiseControlApi> cruiseControlClientProvider;
+
+    private String ccHost = null;
+
+    /**
+     * @param vertx The Vertx instance
+     * @param pfa Platform features availability properties
+     * @param supplier Supplies the operators for different resources
+     */
+    public KafkaRebalanceAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
+                                          ResourceOperatorSupplier supplier) {
+        this(vertx, pfa, supplier, v -> new CruiseControlApiImpl(vertx), null);
+    }
+
+    /**
+     * @param vertx The Vertx instance
+     * @param pfa Platform features availability properties
+     * @param supplier Supplies the operators for different resources
+     * @param ccHost Optional host address for the Cruise Control REST API. If this is not supplied then Cruise Control
+     *             service address will be used. This parameter is intended for use in testing.
```

Comment:
Can you do this by:

* Having a protected method to return the `ccHost()` at the point you need it. Impl in this class returns the service address.
* Use a subclass for the test which returns something else? 

That would be neater than having this field which we don't actually need in the production code.